File: rssreader/src/main/java/com/khalturin/dmitriy/rssreader/di/component/PresenterComponent.java
package com.khalturin.dmitriy.rssreader.di.component;

import com.khalturin.dmitriy.rssreader.di.PerPresenter;
import com.khalturin.dmitriy.rssreader.di.module.PresenterModule;
import com.khalturin.dmitriy.rssreader.view.activity.BookmarksActivity;
import com.khalturin.dmitriy.rssreader.view.activity.FeedActivity;
import com.khalturin.dmitriy.rssreader.view.activity.NewsActivity;

import dagger.Component;

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 14.04.18 17:48.
 */

@PerPresenter
@Component(modules = {
  PresenterModule.class
})
public interface PresenterComponent {

  void inject(FeedActivity activity);
  void inject(NewsActivity activity);
  void inject(BookmarksActivity activity);

}

File: rssreader/src/main/kotlin/com/khalturin/dmitriy/rssreader/di/component/PresenterComponent.kt
package com.khalturin.dmitriy.rssreader.di.component

import com.khalturin.dmitriy.rssreader.di.PerPresenter
import com.khalturin.dmitriy.rssreader.di.module.PresenterModule
import com.khalturin.dmitriy.rssreader.view.activity.ArticleActivity
import com.khalturin.dmitriy.rssreader.view.activity.BookmarksActivity
import com.khalturin.dmitriy.rssreader.view.activity.FeedActivity
import dagger.Component

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 26.10.18 21:45.
 */
@PerPresenter
@Component(
  modules = [
    PresenterModule::class
  ]
)
interface PresenterComponent {

  fun inject(activity: FeedActivity)
  fun inject(activity: ArticleActivity)
  fun inject(activity: BookmarksActivity)

}
